recipes/hdf4/build.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+mkdir -vp ${PREFIX}/bin;
+
+ARCH="$(uname 2>/dev/null)"
+
+export CFLAGS="-m64 -pipe -O2 -march=x86-64 -fPIC"
+export CXXFLAGS="${CFLAGS}"
+#export CPPFLAGS="-I${PREFIX}/include"
+#export LDFLAGS="-L${PREFIX}/lib"
+
+LinuxInstallation() {
+
+    chmod +x configure;
+
+    ./configure \
+        --disable-static \
+        --enable-linux-lfs \
+        --with-ssl \
+        --with-zlib \
+        --prefix=${PREFIX} || return 1;
+    make || return 1;
+    make install || return 1;
+
+    rm -rf ${PREFIX}/share/hdf4_examples;
+
+    return 0;
+}
+
+OSXInstallation() {
+
+    chmod +x configure;
+
+    ./configure \
+        --disable-static \
+        --enable-linux-lfs \
+        --with-ssl \
+        --disable-fortran \
+        --with-zlib \
+        --prefix=${PREFIX} || return 1;
+    make || return 1;
+    make install || return 1;
+
+    rm -rf ${PREFIX}/share/hdf4_examples;
+
+    return 0;
+}
+
+case ${ARCH} in
+    'Linux')
+        LinuxInstallation || exit 1;
+        ;;
+    'Darwin')
+        OSXInstallation || exit 1;
+        ;;
+    *)
+        echo -e "Unsupported machine type: ${ARCH}";
+        exit 1;
+        ;;
+esac
```

Comment:
In general, we try not to break things up like this. We try to have as many commands as possible share by Linux and Mac. Instead, please set options that will be different in some environment variable in an `if` statement before. The rest of this can be the same for both cases.